sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgBuilder_Instructions.cs

```diff
@@ -1596,6 +1596,29 @@ public void Foo(Class1 some, Class1 other)
             return null;
         }
     }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+
+        [TestMethod]
+        public void OtherBinaryExpressions()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        public void Foo(string s)
+        {
+            if (s == null)                  // %0 := string.operator ==(string, string) [ string s const ]
```

Comment:
Hum your question is actually hard to answer because you are talking about 2 things at the same time. Regarding the conditionals/branches, we will generate the branches in the uCFG but the engine doesn't really care whether a path is accessible or not (we only care about whether or not there is a path). Regarding the simple comparison, it was decided to generate the instruction for operators (I need to filter the primitive types).